rackspace/networking/v2/ipaddresses/requests.go

```diff
@@ -0,0 +1,230 @@
+type ListOpts struct{}
+
+// ToSharedIPListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToSharedIPListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), nil
+}
+
+// List returns a Pager which allows you to iterate over a collection of
+// shared IPs.
+func List(c *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(c)
+	if opts != nil {
+		query, err := opts.ToSharedIPListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+
+	return pagination.NewPager(c, url, func(r pagination.PageResult) pagination.Page {
+		return IPAddressPage{pagination.SinglePageBase(r)}
+	})
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToSharedIPCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts contains options for provisioning a shared IP address. This object is passed to
+// the sharedips.Create function.
+type CreateOpts struct {
+	// REQUIRED
+	NetworkID string
+	// REQUIRED
+	PortIDs []string
+	// REQUIRED
+	Version gophercloud.IPVersion
+	// REQUIRED
+	TenantID string
+}
+
+// ToSharedIPCreateMap assembles a request body based on the contents of a
+// CreateOpts.
+func (opts CreateOpts) ToSharedIPCreateMap() (map[string]interface{}, error) {
+	v := make(map[string]interface{})
+
+	if opts.NetworkID == "" {
+		return nil, fmt.Errorf("Required CreateOpts field 'NetworkID' not set.")
+	}
+	v["network_id"] = opts.NetworkID
+
+	if len(opts.PortIDs) < 2 {
+		return nil, fmt.Errorf("Required CreateOpts field 'PortIDs' must contain at least 2 port IDs.")
+	}
+	v["port_ids"] = opts.PortIDs
+
+	if opts.Version != 4 && opts.Version != 6 {
+		return nil, fmt.Errorf("Required CreateOpts field 'Version' not set.")
+	}
+	v["version"] = opts.Version
+
+	if opts.TenantID == "" {
+		return nil, fmt.Errorf("Required CreateOpts field 'TenantID' not set.")
+	}
+	v["tenant_id"] = opts.TenantID
+
+	return map[string]interface{}{"ip_address": v}, nil
+}
+
+// Create will provision a new shapred IP address based on the values in CreateOpts. To extract
+// the IPAddress object from the response, call the Extract method on the
+// CreateResult.
+func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) CreateResult {
+	var res CreateResult
+
+	reqBody, err := opts.ToSharedIPCreateMap()
+	if err != nil {
+		res.Err = err
+		return res
+	}
+
+	_, res.Err = client.Post(createURL(client), reqBody, &res.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{201},
+	})
+	return res
+}
+
+// Delete will deallocate the existing shared IP with the provided ID from the tenant.
+// Before using this operation, all IP associations must be removed from the IP address
+// by using the Disassociate function.
+func Delete(client *gophercloud.ServiceClient, id string) DeleteResult {
```

Comment:
Hmm, I think it should be explicit. The flag is a good idea. Let's add a `DeleteOptsBuilder` parameter that we can add fields to later for this. One of the things I'd want in a Gophercloud v2 is for all the operations to take `*OptsBuilder` parameters even if they currently need it so that we can retroactively add parameters like this.